src/EFCore.PG/Query/Expressions/Internal/PostgresDeleteExpression.cs

```diff
@@ -0,0 +1,73 @@
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.Expressions.Internal;
+
+public sealed class PostgresDeleteExpression : Expression, IPrintableExpression
+{
+    /// <summary>
+    ///     The tables that rows are to be deleted from.
+    /// </summary>
+    public TableExpression Table { get; }
+
+    /// <summary>
+    ///     Additional tables which can be referenced in the predicate.
+    /// </summary>
+    public IReadOnlyList<TableExpression> FromItems { get; }
```

Comment:
The FromItems can be subquery?

Reply:
I'll test whether that's suppotred, thanks.

Reply:
Yep, can be a subquery! Will add support, do you think it's worth adding a test for that in relational?

Reply:
Will add a test in relational in same PR.

Reply:
added test for 5 kind of joins possible

Reply:
Synced to your PR, looks good!